### ISSUE TITLE
[Testerina] NullPointerException when the tests have a service

### ISSUE THREAD

**Issue by MaryamZi** [ballerina-platform/ballerina-lang#21855]
**Description:**
Ballerina build for a simple module with a service in the tests folder results in a NPE.

Following is the test content.
```ballerina
import ballerina/io;
import ballerina/lang.'object;
import ballerina/test;

type Listener object {
    *'object:Listener;

    public function __init() {
    }

    public function __attach(service s, string? name = ()) returns error? {
    }

    public function __detach(service s) returns error? {
    }

    public function __start() returns error? {
    }

    public function __gracefulStop() returns error? {
        return ();
    }

    public function __immediateStop() returns error? {
        return ();
    }
};

service s on new Listener() {
    
}

@test:Config {
    
}
function testFunc() {
    io:println("I'm the test!");
}
```

```cmd
$ ballerina build foo
Compiling source
        maryam/foo:0.1.0

Creating balos
        target/balo/foo-2020r1-any-0.1.0.balo

Running Tests
        maryam/foo:0.1.0
error: java.lang.NullPointerException
error: there are test failures
```

**Affected Versions:**
'stage' branch with the changes in https://github.com/ballerina-platform/ballerina-lang/pull/21841. Without the changes in https://github.com/ballerina-platform/ballerina-lang/pull/21841, the build fails with a bad, sad error at BIRGen.

**Comment by warunalakshitha** [comment]
This is the stacktrace

[2020-03-17 11:16:31,556] SEVERE {b7a.log.crash} - java.lang.NullPointerException
org.ballerinalang.jvm.util.exceptions.BallerinaException: java.lang.NullPointerException
        at org.ballerinalang.test.runtime.entity.TesterinaFunction.runOnSchedule(TesterinaFunction.java:122)
        at org.ballerinalang.test.runtime.entity.TesterinaFunction.invoke(TesterinaFunction.java:53)
        at org.ballerinalang.test.runtime.BTestRunner.startSuite(BTestRunner.java:457)
        at org.ballerinalang.test.runtime.BTestRunner.execute(BTestRunner.java:272)
        at org.ballerinalang.test.runtime.BTestRunner.runTest(BTestRunner.java:97)
        at org.ballerinalang.test.runtime.util.TesterinaUtils.executeTests(TesterinaUtils.java:71)
        at org.ballerinalang.test.runtime.Main.startTestSuit(Main.java:59)
        at org.ballerinalang.test.runtime.Main.main(Main.java:53)
Caused by: java.lang.NullPointerException
        at waruna.mod1.mod1.waruna_mod1_0_1_0__testinit_(mod1:29)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.ballerinalang.test.runtime.entity.TesterinaFunction.lambda$runOnSchedule$0(TesterinaFunction.java:106)
        at org.ballerinalang.jvm.scheduling.SchedulerItem.execute(Scheduler.java:482)
        at org.ballerinalang.jvm.scheduling.Scheduler.run(Scheduler.java:263)
        at org.ballerinalang.jvm.scheduling.Scheduler.runSafely(Scheduler.java:231)
        at java.lang.Thread.run(Thread.java:748)

**Comment by MaryamZi** [comment]
Getting the same error when the source itself has an service where the listener it is attached to is from an imported module.

That does't fail on RC1 though.

**Comment by MaryamZi** [comment]
> Getting the same error when the source itself has an service where the listener it is attached to is from an imported module.
> 
> That does't fail on RC1 though.

This second case seems to be working on RC2.

**Comment by MaryamZi** [comment]
Fixed with #21961.
